lib/httplib/httpheaders.go

```diff
@@ -63,51 +115,39 @@ func SetDefaultSecurityHeaders(h http.Header) {
 
 // SetIndexContentSecurityPolicy sets the Content-Security-Policy header for main index.html page
 func SetIndexContentSecurityPolicy(h http.Header) {
-	var cspValue = strings.Join([]string{
-		GetDefaultContentSecurityPolicy(),
-		// 'unsafe-inline' is required by CSS-in-JS to work
-		"style-src 'self' 'unsafe-inline'",
-		"img-src 'self' data: blob:",
-		"font-src 'self' data:",
-		"connect-src 'self' wss:",
-	}, ";")
-
-	h.Set("Content-Security-Policy", cspValue)
+	cspString := getContentSecurityPolicyString(
+		defaultContentSecurityPolicy,
+		defaultFontSrc,
+		map[string]string{
+			"connect-src": "'self' wss:",
+		},
+	)
+
+	h.Set("Content-Security-Policy", cspString)
 }
 
 // SetAppLaunchContentSecurityPolicy sets the Content-Security-Policy header for /web/launch
 func SetAppLaunchContentSecurityPolicy(h http.Header, applicationURL string) {
-	var cspValue = strings.Join([]string{
-		GetDefaultContentSecurityPolicy(),
-		// 'unsafe-inline' is required by CSS-in-JS to work
-		"style-src 'self' 'unsafe-inline'",
-		"img-src 'self' data: blob:",
-		"font-src 'self' data:",
-		fmt.Sprintf("connect-src 'self' %s", applicationURL),
-	}, ";")
-
-	h.Set("Content-Security-Policy", cspValue)
+	cspString := getContentSecurityPolicyString(
+		defaultContentSecurityPolicy,
+		defaultFontSrc,
+		map[string]string{
+			"connect-src": "'self' " + applicationURL,
+		},
+	)
+
+	h.Set("Content-Security-Policy", cspString)
 }
 
-// GetDefaultContentSecurityPolicy provides a starting Content Security Policy with safe defaults.
-func GetDefaultContentSecurityPolicy() string {
-	return strings.Join([]string{
-		"default-src 'self'",
-		// specify CSP directives not covered by `default-src`
-		"base-uri 'self'",
-		"form-action 'self'",
-		"frame-ancestors 'none'",
-		// additional default restrictions
-		"object-src 'none'",
-		// auto-pay plans in Cloud use stripe.com to manage billing information
-		"script-src 'self' https://js.stripe.com",
-		"frame-src https://js.stripe.com",
-	}, ";")
-}
+func SetRedirectPageContentSecurityPolicy(h http.Header, scriptSrc string) {
+	cspString := getContentSecurityPolicyString(
+		defaultContentSecurityPolicy,
+		map[string]string{
+			"script-src": fmt.Sprintf("'%s'", scriptSrc),
```

Comment:
```suggestion
			"script-src": "'" + scriptSrc + "'",
```

nit: doesn't really matter, but it would be nice if we reserved sprintf for more complicated formatting and didn't use it for simpler concatenation operations